File: benchmark/tasks/refactor-legacy/verify.mjs
import { pathToFileURL } from "node:url";
import path from "node:path";
import assert from "node:assert/strict";
import { access } from "node:fs/promises";

const workspace = process.argv[2];
if (workspace === undefined) {
    throw new Error("缺少 workspace 参数");
}

// legacy.js 应被删除
await assert.rejects(access(path.resolve(workspace, "src/legacy.js")), "legacy.js 应已删除");

const { sumPair } = await import(pathToFileURL(path.resolve(workspace, "src/use-a.js")).href);
const { double } = await import(pathToFileURL(path.resolve(workspace, "src/use-b.js")).href);
const { outerProduct } = await import(pathToFileURL(path.resolve(workspace, "src/use-c.js")).href);

assert.equal(sumPair(2, 3), 5);
assert.equal(double(4), 8);
assert.deepEqual(outerProduct(2, 3), [5, 6]);

console.log("refactor-legacy 验收通过");

File: benchmark/tasks/refactor-legacy/verify.py
import pathlib
import sys

workspace = pathlib.Path(sys.argv[1])
sys.path.insert(0, str(workspace))

assert not (workspace / "src" / "legacy.py").exists(), "legacy.py 应已删除"
from src.use_a import sum_pair
from src.use_b import double
from src.use_c import outer_product

assert sum_pair(2, 3) == 5
assert double(4) == 8
assert outer_product(2, 3) == [5, 6]

for path in (workspace / "src").glob("*.py"):
    if "legacy_calc" in path.read_text(encoding="utf-8"):
        raise AssertionError(f"{path.name} 仍然引用 legacy_calc")

print("refactor-legacy 验收通过")
